CorsixTH/Src/main.cpp

```diff
@@ -52,6 +65,60 @@ inline void preload_lua_package(lua_State* L, const char* name,
       fn);
 }
 
+std::string search_script_file(lua_State* L) {
+  // 1. Check for --interpreter
+  int iNArgs = lua_gettop(L);
+  for (int i = 1; i <= iNArgs; ++i) {
+    if (lua_type(L, i) == LUA_TSTRING) {
+      size_t iLen;
+      const char* sCmd = lua_tolstring(L, i, &iLen);
+      if (iLen > 14 && std::memcmp(sCmd, "--interpreter=", 14) == 0)
+        return sCmd + 14;
+    }
+  }
+
+#ifdef CORSIX_SEARCH_LOCAL_DATADIRS
+  // 2. Find CorsixTH.lua in working dir and program dir
+  static constexpr std::array<const char*, 3> asSearchDirs{
```

Comment:
```suggestion
  static constexpr std::array<const char*, 4> asSearchDirs{
```
With that, approved.

Reply:
@tobylane have you tested it?

Reply:
Yes, I can move it and run it.